File: CPP/ByRound/1743/1743_A_Password.cpp
// Problem: CF 1743 A - Password
// https://codeforces.com/contest/1743/problem/A

#include <cstdio>
int T, n;
int main() {
  // Flowerbox:
  // Purpose: Calculate the number of valid 4-digit passwords under constraints.
  // Algorithm: The password uses exactly two distinct digits, each appearing exactly twice.
  // We are given n digits that are definitely NOT used in the password.
  // Available digits = 10 - n
  // Number of ways to pick 2 distinct digits from available = C(available, 2)
  // Each pair can be arranged in 4!/(2!*2!) = 6 ways (since we have two pairs of identical digits)
  // So total = C(available, 2) * 6
  // Time Complexity: O(1) per test case (constant time operations)
  // Space Complexity: O(1) (only using a few variables)
  
  for (scanf("%d", &T); T--;)
    for (scanf("%d", &n), printf("%d\n", 3 * (10 - n) * (9 - n)); n--;)
      scanf("%*d");
}


// https://github.com/VaHiX/codeForces/

File: Python/ByTier/A/1743_A_Password.py
# Problem: CF 1743 A - Password
# https://codeforces.com/contest/1743/problem/A

# Flowerbox:
# Purpose: Calculate the number of valid 4-digit passwords where exactly two different digits appear,
#          each digit appearing exactly twice. Given a list of unused digits, find how many valid
#          4-digit combinations exist.
#
# Algorithm: For each test case:
#            1. Read n (number of unused digits) and the set of unused digits.
#            2. Generate all possible pairs of distinct digits from the remaining 10 - n digits.
#            3. For each such pair, calculate the number of ways to arrange them in a 4-digit password
#               where each digit appears exactly twice (i.e., choose positions for first digit: C(4,2) = 6).
#            4. Total is sum over all valid pairs.
#
# Time Complexity: O(n^2) per test case due to generating pairs of digits from available digits.
# Space Complexity: O(1), constant extra space (excluding input/output).

print("\n".join([str(3 * (10 - n) * (9 - n)) for n in map(int, [*open(0)][1::2])]))


# https://github.com/VaHiX/codeForces/